components/list/list-item-drag-mixin.js

```diff
@@ -33,86 +95,299 @@ export const ListItemDragMixin = superclass => class extends superclass {
 		}
 	}
 
-	moveAfter(itemKey) {
+	moveItemAfterMe(itemKey) {
 		if (itemKey === undefined) {
 			return;
 		}
-		this._dispatchDragEvent(itemKey);
+		this._moveItem(itemKey, this.key, false);
 	}
 
-	moveBefore(itemKey) {
+	moveItemBeforeMe(itemKey) {
 		if (itemKey === undefined) {
 			return;
 		}
 
-		this._dispatchDragEvent(itemKey);
+		this._moveItem(itemKey, this.key, true);
+	}
+
+	_dragAreaClick(e) {
+		this.shadowRoot.querySelector(`#${this._itemDragId}`).activateKeyboardMode();
+		e.preventDefault();
 	}
 
-	// TODO
-	_copy() {
+	_dragBottomEnter(e) {
+		e.dataTransfer.dropEffect = 'move';
+		const dropSpots = dropSpotsFactory();
+		dropSpots.setDestination(this, move.below);
+		this._inBottomArea = true;
+	}
 
+	_dragEnd(e) {
+		const dropSpot = dropSpotsBlowUp();
+		this.beingDragged = false;
+		this.tilt = false;
+		if (dropSpot.shouldDrop(e.timeStamp)) {
+			this._moveItem(dropSpot.targetKey, dropSpot.destinationKey, dropSpot.moveBeforeDestination);
+		}
 	}
 
-	_dispatchDragEvent(detail) {
+	_dragHandleActions(e) {
+		let destinationKey;
+		switch (e.detail.action) {
+			case dragActions.active:
+				this._keyboardMode = true;
+				break;
+			case dragActions.cancel:
+			case dragActions.save:
+				this._keyboardMode = false;
+				break;
+			case dragActions.up:
+				destinationKey = this.previousElementSibling && this.previousElementSibling.key;
```

Comment:
I will be modifying the object sent by the event in a way where we can remove the need to know about our siblings! This might come as a separate PR. Tight deadline and this works? Tests are more important and I will see if I get to this.